imsms_analysis/analysis_runner.py

```diff
@@ -73,6 +77,105 @@ def run(self, factory):
+                (
+                final_biom, 
+                target, 
+                _, _
+                ) = run_preprocessing(config, self.callbacks)
+
+                base_dir = "dataset"
+                dataset = "imsms-mlab"
+                preparation = config.analysis_name
+                target_name = "disease_binary"
+                algorithm = config.mlab_algorithm
+                if algorithm is None:
+                    algorithm = "RandomForestClassifier"
+
+                # Create the  expected file structure
+                results_dir = path.join(
+                    base_dir, dataset, preparation, target_name
+                )
+                if not path.exists(results_dir):
+                    makedirs(results_dir)
+                
+                # Save table and metadata
+                table_fp = path.join(
+                    base_dir, dataset, preparation, target_name,
+                    "filtered_table.qza"
+                )
+                table_artifact = Artifact.import_data(
+                    "FeatureTable[Frequency]", 
+                    final_biom
+                )
+                table_artifact.save(table_fp)
+
+                metadata_fp = path.join(
+                    base_dir, dataset, preparation, target_name,
+                    "filtered_metadata.qza"
+                )
+                metadata_artifact = Artifact.import_data(
+                    "SampleData[Target]", target
+                )
+                metadata_artifact.save(metadata_fp)
+                
+
+                # run job indices 1...10 inclusive, letting 5 jobs run at once
+                # each job has 100 parameter sets, for a total of 1000 parameter sets
+                start = 1
+                end = 10
+                n_concurrent_jobs = 5
+                chunk_size = 100
+                (
+                    script_fp,
+                    params_fp,
+                    run_info_fp,
+                ) = orchestrate_hyperparameter_search(
+                    dataset=dataset,
+                    preparation=preparation,
+                    target=target_name,
+                    algorithm=algorithm,
+                    repeats=3,  # num CV repeats
+                    base_dir=base_dir,  # Directory with mlab structure containing datasets
+                    ppn=4,  # processors per node
+                    memory=32,  # memory in GB
+                    wall=50,  # walltime  in hours
+                    chunk_size=chunk_size,  # num parameter sets to run per job
+                    randomize=True,  # randomly shuffle order of parameter set
+                    force=False,  # force overwrite of existing results
+                    dry=False,  # dry runs
+                    dataset_path=table_fp,
+                    metadata_path=metadata_fp,
+                )
+                
+                cmd = ["qsub", "-t", f"{start}-{end}%{n_concurrent_jobs}", script_fp]
```

Comment:
Adding a note here to remind myself to add logic that caps the value of `end` at `num_parameter_sets/chunk_size` so we're not submitting jobs for parameter sets that don't exist. I usually do this when I submit jobs for algorithms with a small number of parameter sets (e.g. LinearSVR/SVC has only 16)